### PR TITLE
Fix numpy deprecations that are errors in 1.24.2

### PR DESCRIPTION
**Pull request by ocefpaf**
I also run pre-commit on the files I touched so we slowly get them all back into compliance.

### CODE REVIEW

**Comment by ocefpaf** [comment]
The integration tests have been broken for quite a while so I won't touch those now. @benjwadams this one is ready for review and merge. It will unblock #1008.

**Comment by ocefpaf** [comment]
@benjwadams please let me know if you have cycles to review this one. It is not a bug change but it is blocking other PRs. The pre-commit failures are already addressed in #1010 and I want to tackle the integration test after this one is merged.

**Review by benjwadams — APPROVED** [review]

**Comment by ocefpaf** [comment]
> SOS datasets.

We should check with folks but my feeling is that we could probably drop the SOS one and mark then as no longer supported. I need to check what they are testing though to be sure.

**Pull request merged**
